templates/systemd-system-prometheus-postgres-exporter-service.j2

```diff
@@ -3,7 +3,7 @@ Description=Prometheus Postgres Exporter Server
 
 [Service]
 User=postgres
-Environment="DATA_SOURCE_NAME=user=postgres dbname={{ prometheus_postgres_dbname }} host=/var/run/postgresql/ sslmode=disable"
+Environment="DATA_SOURCE_NAME=postgresql://{{ prometheus_postgres_user }}:{{ prometheus_postgres_user_pass }}@localhost:5432/{{ prometheus_postgres_dbname }}?sslmode=disable"
```

Comment:
We need to ensure this remains backwards compatible. It's probably easier if you make the whole data source name into a variable e.g.
`prometheus_postgres_data_source_name: "user=postgres dbname={{ prometheus_postgres_dbname }} host=/var/run/postgresql/ sslmode=disable"`
in defaults.yml, then you've got full flexibility to override it completely.

Reply:
thank you for pointing out the backward compatibility guarantee. I feel a long configuration line is not really friendly so I just put another variable(password) in DATA_SOURCE_NAME.

Reply:
For your requirements I can see why it seems unfriendly, but this line currently includes other parameters such as `sslmode=disable` and support for a remote host. This increases the likelihood that other parameters will be necessary, so making the who line configurable would avoid future changes.